src/main/java/net/minecraftforge/common/DimensionManager.java

```diff
@@ -216,20 +216,24 @@ public static void initDimension(int dim)
         {
             throw new RuntimeException("Cannot Hotload Dim: Overworld is not Loaded!");
         }
+        WorldServer alterantiveDimension = null;
         try
         {
-            DimensionManager.getProviderType(dim);
+            DimensionType type = DimensionManager.getProviderType(dim);
+            if (dim != 0) {
+                DimensionPreloadEvent adEvent = new DimensionPreloadEvent(dim, type);
+                MinecraftForge.EVENT_BUS.post(adEvent);
+                alterantiveDimension = adEvent.getAlternativeDimension();
+            }
         }
         catch (Exception e)
         {
             System.err.println("Cannot Hotload Dim: " + e.getMessage());
             return; // If a provider hasn't been registered then we can't hotload the dim
         }
         MinecraftServer mcServer = overworld.getMinecraftServer();
-        ISaveHandler savehandler = overworld.getSaveHandler();
-        //WorldSettings worldSettings = new WorldSettings(overworld.getWorldInfo());
 
-        WorldServer world = (dim == 0 ? overworld : (WorldServer)(new WorldServerMulti(mcServer, savehandler, dim, overworld, mcServer.theProfiler).init()));
+        WorldServer world = (dim == 0 ? overworld : ( alterantiveDimension != null ? alterantiveDimension : ((WorldServer)(new WorldServerMulti(mcServer, overworld.getSaveHandler(), dim, overworld, mcServer.theProfiler).init()))));
```

Comment:
This line has two ternary statements in it now, can you break it up into if / else if / else?